### PR TITLE
Add vita environment

### PR DESCRIPTION
**Pull request by d3m3vilurr**
from [xyzz/moonlight-common-c](https://github.com/xyzz/moonlight-common-c)

### CODE REVIEW

File: src/PlatformSockets.c [hunk 1]
```diff
@@ -80,6 +95,9 @@ SOCKET bindUdpSocket(int addrfamily, int bufferSize) {
     int err;
 
     LC_ASSERT(addrfamily == AF_INET || addrfamily == AF_INET6);
+#ifdef __vita__
+    addrfamily = AF_INET;
```
Comment (cgutman) [inline src/PlatformSockets.c]: Why is this needed? I don't think it will ever attempt to bind an IPv6 address unless the address resolves to IPv6 (which it won't because you're using gethostbyname()).

File: src/PlatformSockets.c [hunk 2]
```diff
@@ -6,6 +10,16 @@
 
 void addrToUrlSafeString(struct sockaddr_storage* addr, char* string)
 {
+    char addrstr[48];
+#ifdef __vita__
+    struct sockaddr_in* sin = (struct sockaddr_in*)addr;
+    sceNetInetNtop(addr->ss_family, &sin->sin_addr, addrstr, sizeof(addrstr));
```
Comment (cgutman) [inline src/PlatformSockets.c]: Can you just define inet_ntop to sceNetInetNtop in PlatformSockets.h like you do with other functions?

File: src/Platform.c [hunk 3]
```diff
@@ -162,6 +202,12 @@ int PltCreateEvent(PLT_EVENT* event) {
     }
 
     return 0;
+#elif defined(__vita__)
+    event->mutex = sceKernelCreateMutex("", 0, 0, NULL);
+    event->cond = sceKernelCreateCond("", 0, event->mutex, NULL);
+    event->signalled = 0;
+    printf("mutex: 0x%x cond: 0x%x\n", event->mutex, event->cond);
```
Comment (cgutman) [inline src/Platform.c]: Can you remove this printf?

File: src/Connection.c [hunk 4]
```diff
@@ -187,6 +188,21 @@ static int resolveHostName(const char* host)
 
     freeaddrinfo(res);
     return 0;
+#else
+    struct hostent *phost = gethostbyname(host);
```
Comment (cgutman) [inline src/Connection.c]: getaddrinfo() just isn't implemented on the Vita?

File: src/Platform.c [hunk 5]
```diff
@@ -61,6 +73,8 @@ int PltCreateMutex(PLT_MUTEX* mutex) {
         return -1;
     }
     return 0;
+#elif defined(__vita__)
+    *mutex = sceKernelCreateMutex("", 0, 0, NULL);
```
Comment (cgutman) [inline src/Platform.c]: Missing *mutex check and return here.

File: src/SdpGenerator.c [hunk 6]
```diff
@@ -308,7 +308,11 @@ char* getSdpPayloadForStreamConfig(int rtspClientVersion, int* length) {
     char* payload;
     char urlSafeAddr[URLSAFESTRING_LEN];
 
+    #ifdef __vita__
```
Comment (cgutman) [inline src/SdpGenerator.c]: Is this still needed?

File: src/Platform.c [hunk 7]
```diff
@@ -139,6 +167,18 @@ int PltCreateThread(ThreadEntry entry, void* context, PLT_THREAD* thread) {
             return -1;
         }
     }
+#elif defined(__vita__)
+    {
+        thread->alive = 1;
+        thread->context = ctx;
+        ctx->thread = thread;
+        thread->handle = sceKernelCreateThread("", ThreadProc, 0, 0x10000, 0, 0, NULL);
```
Comment (cgutman) [inline src/Platform.c]: Is there no pthreads port to Vita?

File: src/Platform.c (hunk 3 above)
Comment (cgutman) [inline src/Platform.c]: It also looks like you should check if sceKernelCreateMutex or sceKernelCreateCond returned null.

File: src/SdpGenerator.c (hunk 6 above)
Comment (d3m3vilurr) [inline src/SdpGenerator.c]: removed

File: src/Connection.c (hunk 4 above)
Comment (d3m3vilurr) [inline src/Connection.c]: getaddrinfo not implemented yet...

File: src/Platform.c (hunk 3 above)
Comment (d3m3vilurr) [inline src/Platform.c]: fixed

File: src/Platform.c (hunk 7 above)
Comment (d3m3vilurr) [inline src/Platform.c]: currently vita newlib pthread looks little buggy. (also i will more check them)

File: src/Platform.c (hunk 5 above)
Comment (d3m3vilurr) [inline src/Platform.c]: fixed

File: src/PlatformSockets.c (hunk 2 above)
Comment (d3m3vilurr) [inline src/PlatformSockets.c]: fixed

File: src/Platform.c [hunk 8]
```diff
@@ -75,6 +75,10 @@ int PltCreateMutex(PLT_MUTEX* mutex) {
     return 0;
 #elif defined(__vita__)
     *mutex = sceKernelCreateMutex("", 0, 0, NULL);
+    if (!*mutex < 0) {
```
Comment (cgutman) [inline src/Platform.c]: Looks like that ! is not supposed to be there.

File: src/PlatformSockets.c (hunk 1 above)
Comment (d3m3vilurr) [inline src/PlatformSockets.c]: i guess vita system not support ipv6. 
so want to prevent ipv6 protocol. (actually it originally xyzz's works)
now i change to `LC_ASSERT` like this cases (but it can make side effect on release mode.)

File: src/PlatformSockets.c [hunk 9]
```diff
@@ -10,24 +6,18 @@
...
-        sprintf(string, "[%s]", addrstr);
-    } else {
-        sprintf(string, "%s", addrstr);
-    }
-#else
     char addrstr[INET6_ADDRSTRLEN];
 
     if (addr->ss_family == AF_INET6) {
+#ifdef __vita__
+        LC_ASSERT(FALSE);
```
Comment (cgutman) [inline src/PlatformSockets.c]: Unless you have strong objections, I'd prefer to remove the #ifdef __vita__ and Vita-only LC_ASSERTs from the IPv6 related code paths in this file. As long as you're using an IPv4-only DNS lookup routine like gethostbyname(), you'll never get IPv6 addresses anyway. I'd like to minimize the Vita-specific code (since I won't be able to easily test it, among other reasons).

File: src/Platform.c (hunk 8 above)
Comment (d3m3vilurr) [inline src/Platform.c]: haha....i'm idiot :(
fixed.

File: src/PlatformSockets.c (hunk 9 above)
Comment (d3m3vilurr) [inline src/PlatformSockets.c]: right.
i just split `sockaddr_in` & `sockaddr_in6` cases.
least, we need this.
cuz sockadder_in6 not defined both vita-header & vita-newlib :)

File: src/PlatformSockets.c [hunk 10]
```diff
@@ -10,14 +10,15 @@ void addrToUrlSafeString(struct sockaddr_storage* addr, char* string)
 
     if (addr->ss_family == AF_INET6) {
 #ifdef __vita__
-        LC_ASSERT(FALSE);
+        struct sockaddr_in* sin = (struct sockaddr_in*)addr;
+        inet_ntop(addr->ss_family, &sin->sin_addr, addrstr, sizeof(addrstr));
```
Comment (cgutman) [inline src/PlatformSockets.c]: In PlatformSockets.h, you've already defined sockaddr_in6 to sockaddr_in and sin6_port to sin_port. I think if you also define sin6_addr to sin_addr, you can remove this `#ifdef`

Comment (d3m3vilurr) [inline src/PlatformSockets.c]: actually these defines are wrong.
in my test, they cause compile error without ifdef macro

```
/home/d3m3vilurr/works/vita/vita-moonlight/third_party/moonlight-common-c/src/PlatformSockets.c:13:41: error: 'struct sockaddr_in' has no member named 'sin6_addr'; did you mean 'sin_addr'?
         inet_ntop(addr->ss_family, &sin6->sin6_addr, addrstr, sizeof(addrstr));
```

so i'll fix them to right way at tonight.

**Comment by cgutman** [comment]
Thanks for your work getting this and the enet changes upstreamed.
